### ISSUE TITLE
[Select] Some options in Group story are not announced as 'selected' when selected on Safari

### ISSUE THREAD

**Issue by a11ymiko** [sl-design-system/components#3431]
### 😯 Current Behavior

For Groups story when user select options from groups 'Happy' or 'Sad' then selected option does not have additional announcement 'Selected' read by VoiceOver. This happens only for options in groups that have header.
This happens only on Safari.

https://github.com/user-attachments/assets/b50572bf-2a96-4c19-b9bd-0ca4b9997435

### 🤔 Expected Behavior/possible solution?

All options should be announced as 'selected' the same way it currently works for options in groups without group header.

### 🖥️ Steps to Reproduce

1. In Safari go to https://storybook.sanomalearning.design/?path=/story/form-select--groups&globals=backgrounds.value:light
2. With VoiceOver turned on select option from 'Happy' group
3. Check if this option is announced as 'selected'
4. Repeat 2. and 3. for option in 'Sad' group

### Were you able to reproduce this in our storybook?

Yes

### If yes, provide the storybook URL

https://storybook.sanomalearning.design/?path=/story/form-select--groups&globals=backgrounds.value:light

### ⚖️ Impact

Not a huge problem, we have a workaround

### Framework

Other

### What browsers are you seeing the problem on?

Safari

### Component version

2.1.4

### 👤 Your name

Kacper

### 🧢 Your product/team

SLDS

### 🎨 Your Theme(s)

Sanoma Learning

**Comment by a11ymiko** [comment]
This was fixed with other PR
